File: projects/GPL-FH-Java/features/Connected/GPL/Graph.java
package GPL;

// *****************************************************************
   
public class Graph 
{
    // Executes Connected Components
    public void run( Vertex s )
    {
	     	System.out.println("Connected");
        ConnectedComponents( );
        original( s );
    }

    public void ConnectedComponents( ) 
    {
        GraphSearch( new RegionWorkSpace( ) );
    }
}

File: projects/GPL-FH-CSharp/features/Cycle/Graph.cs
/*layer Cycle;

import java.lang.Integer;
*/
// Cycle checking, Edge-Neighbor implementation
  
// *************************************************************************
   
public class Graph {//refines

    // Executes Cycle Checking
    public virtual void run( Vertex s )
     {
        System.Console.Out.WriteLine( "Cycle? " + CycleCheck() );
        original( s );
    }
              
    public virtual bool CycleCheck() {
        CycleWorkSpace c = new CycleWorkSpace( isDirected );
        GraphSearch( c );
        return c.AnyCycles;
    }
}
